kopykat-ksp/src/main/kotlin/at/kopyk/KopyKatProcessor.kt

```diff
@@ -51,4 +54,18 @@ internal class KopyKatProcessor(
       }
     }
   }
+
+  @OptIn(KspExperimental::class)
+  private fun KSClassDeclaration.isAnnotated(): Boolean = when {
```

Comment:
You're 100% correct, I've refactored the code in the way you describe.

I've also changed the property to `generateAll`, so the default is `true`, as with all the other options. I've dropped `annotatedOnly`, though, because I think `generateAll = false` is good enough to describe the situation.